tests/integration/test_ingress_traefik.py

```diff
@@ -2,68 +2,50 @@
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-import asyncio
+
 import json
 
-import pytest
 import requests
-import sh
-
-from . import ZINC
 
-TRAEFIK = "traefik-k8s"
+from . import TRAEFIK, ZINC
+from .juju import juju, retry, run_action, status, wait_for_idle
 
 
-@pytest.mark.abort_on_fail
-async def test_ingress_traefik_k8s(ops_test, zinc_charm, zinc_oci_image):
+def test_deploy(zinc_charm, zinc_oci_image):
     """Test that Zinc can be related with Traefik for ingress."""
     apps = [ZINC, TRAEFIK]
-
-    await asyncio.gather(
-        ops_test.model.deploy(
-            zinc_charm,
-            resources={"zinc-image": zinc_oci_image},
-            application_name=ZINC,
-        ),
-        ops_test.model.deploy(
-            TRAEFIK,
-            application_name=TRAEFIK,
-            channel="edge",
-            config={"routing_mode": "subdomain", "external_hostname": "foo.bar"},
-            trust=True,
-        ),
-        ops_test.model.wait_for_idle(apps=apps, status="active", timeout=1000),
+    juju("deploy", zinc_charm, ZINC, "--resource", f"zinc-image={zinc_oci_image}")
+    juju(
+        "deploy",
+        TRAEFIK,
+        "--channel=latest/edge",
+        "--config",
+        "routing_mode=subdomain",
+        "--config",
+        "external_hostname=foo.bar",
+        "--trust",
     )
 
-    # Create the relation
-    await ops_test.model.integrate(ZINC, TRAEFIK)
-    # Wait for the two apps to quiesce
-    await ops_test.model.wait_for_idle(apps=apps, status="active", timeout=1000)
+    juju("integrate", *apps)
+    wait_for_idle(apps)
 
-    result = await _retrieve_proxied_endpoints(ops_test, TRAEFIK)
-    assert result.get(ZINC, None) == {"url": f"http://{ops_test.model_name}-{ZINC}.foo.bar/"}
 
+def test_ingress_setup():
+    """Test that Zinc/Traefik are configured correctly."""
+    model_name = status()["model"]["name"]
+    result = run_action(f"{TRAEFIK}/0", "show-proxied-endpoints")
+    j = json.loads(result["proxied-endpoints"])
+    assert j[ZINC] == {"url": f"http://{model_name}-{ZINC}.foo.bar/"}
```

Comment:
not clear why this refactoring should happen as part of this PR which targets removal of pytest-operator